frontend/src/app/shared/components/storages/upload/sharepoint-upload.strategy.ts

```diff
@@ -0,0 +1,82 @@
+//-- copyright
+// OpenProject is an open source project management software.
+// Copyright (C) the OpenProject GmbH
+//
+// This program is free software; you can redistribute it and/or
+// modify it under the terms of the GNU General Public License version 3.
+//
+// OpenProject is a fork of ChiliProject, which is a fork of Redmine. The copyright follows:
+// Copyright (C) 2006-2013 Jean-Philippe Lang
+// Copyright (C) 2010-2013 the ChiliProject Team
+//
+// This program is free software; you can redistribute it and/or
+// modify it under the terms of the GNU General Public License
+// as published by the Free Software Foundation; either version 2
+// of the License, or (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program; if not, write to the Free Software
+// Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+//
+// See COPYRIGHT and LICENSE files for more details.
+//++
+
+import { Observable } from 'rxjs';
+import { map, share } from 'rxjs/operators';
+import { HttpClient, HttpEvent } from '@angular/common/http';
+
+import { IUploadFile } from 'core-app/core/upload/upload.service';
+import { EXTERNAL_REQUEST_HEADER } from 'core-app/features/hal/http/openproject-header-interceptor';
+import { IUploadStrategy } from 'core-app/shared/components/storages/upload/upload-strategy';
+import convertHttpEvent from 'core-app/core/upload/convert-http-event';
+
+export interface SharepointFileUploadResponse {
+  id:string;
+  name:string;
+  file:{ mimeType:string };
+  size:number;
+}
+
+export class SharepointUploadStrategy implements IUploadStrategy {
+  constructor(private readonly http:HttpClient) { }
```

Comment:
I'm also getting this error here, which is first wrong (http is used in line 58) and I don't know how to get rid off.

Reply:
Uh, does JS/TS have syntactic sugar for

```
this.foo = foo;
```

?

That's neat. Is it new enough that ESLint does legitimately not understand this? Then maybe an update could help. Other than that, I have no clue :see_no_evil:

Reply:
It possible that you might need to define the `http` class member for eslint to pick up on it

```
export class SharepointUploadStrategy implements IUploadStrategy {
  private readonly http:HttpClient // <----

  constructor(private readonly http:HttpClient) { }
//....
}
```

Reply:
I somehow think, that is a red herring. I do not see the same linting issue, when running eslint locally and it is clearly used in line 58.

I'd vote for not letting it block this PR. But to have a look in a separate implementation chunk. I'd love to investigate this, but after wednesday